components/autocomplete/index.js

```diff
@@ -231,18 +232,26 @@ export class Autocomplete extends Component {
 		}
 	}
 
-	loadOptions( index ) {
-		this.props.completers[ index ].getOptions().then( ( options ) => {
-			const keyedOptions = map( options, ( option, i ) => ( { ...option, key: index + '-' + i } ) );
-			const filteredOptions = filterOptions( this.state.search, keyedOptions );
-			const selectedIndex = filteredOptions.length === this.state.filteredOptions.length ? this.state.selectedIndex : 0;
-			this.setState( {
-				[ 'options_' + index ]: keyedOptions,
-				filteredOptions,
-				selectedIndex,
+	/**
+	 * Load options for an autocompleter.
+	 *
+	 * @param {number} index The autocompleter index.
+	 * @param {string} query The query, if any.
+	 */
+	loadOptions( index, query ) {
+		this.props.completers[ index ]
+			.getOptions( query )
+			.then( ( options ) => {
+				const keyedOptions = map( options, ( option, i ) => ( { ...option, key: index + '-' + ( option.value.slug ? option.value.slug : i ) } ) );
```

Comment:
What is `slug` here? Are we assuming a particularly-shaped REST API entity where the autocomplete options are otherwise generic?

Reply:
Hmmm, I'm not sure we need this maybe a remnant of a previous approach, I'll revert this change